src/pal/src/misc/miscpalapi.cpp

```diff
@@ -298,3 +301,153 @@ PAL_Random(
     PERF_EXIT(PAL_Random);
     return bRet;
 }
+
+
+// Support for EnsureOpenSslInitialized
+
+static const char * const libcryptoName = "libcrypto" PAL_SHLIB_SUFFIX;
```

Comment:
Well, this is where things get tricky :)  Different versions of linux can and will have differing versions of the library (see the latest openssl is 1.0.something now) and there is no guarantee based on 'Linux' as to what is available.

There is no 'dlopen the installed version of this library' API as well, so we probably need to think about exactly what we want to do here.  I can think of a few options right now, but none of them are great.  I'll spend some time thinking about it, but we should open up the discussion as well.